src/utils/ucc_parser.c

```diff
@@ -6,7 +6,159 @@
 #include "ucc_parser.h"
 #include "ucc_malloc.h"
 #include "ucc_log.h"
-#include "khash.h"
+#include "ucc_string.h"
+#include "ini.h"
+
+#define UCC_ADD_KEY_VALUE_TO_HASH(_type, _h, _name, _val)                \
+    do {                                                                 \
+        khiter_t _iter;                                                  \
+        int      _res;                                                   \
+        char    *_dup;                                                   \
+        _iter = kh_get(_type, _h, _name);                                \
+        if (_iter != kh_end(_h)) {                                       \
+            ucc_warn("found duplicate '%s' in config file", _name);      \
+            return 0;                                                    \
+        } else {                                                         \
+            _dup = strdup(_name);                                        \
+            if (!_dup) {                                                 \
+                ucc_error("failed to dup str for kh_put");               \
+                return 0;                                                \
+            }                                                            \
+            _iter = kh_put(_type, _h, _dup, &_res);                      \
+            if (_res == UCS_KH_PUT_FAILED) {                             \
+                ucc_free(_dup);                                          \
+                ucc_error("inserting '%s' to config map failed", _name); \
+                return 0;                                                \
+            }                                                            \
+        }                                                                \
+        _dup = strdup(_val);                                             \
```

Comment:
It is not a memory leak - memory still in use as long as hash table is in use. It will be freed in function ucc_release_file_config (line 393 in ucc_parser.c)

Reply:
got it, thanks